test/cpp/util/spiffe_test_credentials.cc

```diff
@@ -0,0 +1,84 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "test/cpp/util/spiffe_test_credentials.h"
+#include "test/core/end2end/data/ssl_test_data.h"
+
+namespace grpc {
+namespace testing {
+
+class TestTlsCredentialReloadInterface
+    : public ::grpc_impl::experimental::TlsCredentialReloadInterface {
+  // Sync implementation.
+  int Schedule(
+      ::grpc_impl::experimental::TlsCredentialReloadArg* arg) override {
+    struct ::grpc_impl::experimental::TlsKeyMaterialsConfig::PemKeyCertPair
+        pem_key_cert_pair = {test_server1_key, test_server1_cert};
+    arg->set_pem_root_certs(test_root_cert);
+    arg->add_pem_key_cert_pair(pem_key_cert_pair);
+    return 0;
+  }
+};
+
+class TestTlsServerAuthorizationCheckInterface
+    : public ::grpc_impl::experimental::TlsServerAuthorizationCheckInterface {
+  // Sync implementation.
```

Comment:
Can we test async version of server authz check like we did in `h2_spiffe.cc`? By testing that, we will have a more confidence on our support to users' async callbacks.

Reply:
Great! Thanks much for adding async test.